CHANGELOG.md

```diff
@@ -47,6 +56,60 @@ Main (unreleased)
+  - [`otelcol.processor.tail_sampling`] Fix precedence of inverted match in and policy.
+    Previously if the decision from a policy evaluation was `NotSampled` or `InvertNotSampled` 
+    it would return a `NotSampled` decision regardless, effectively downgrading the result.
+    This was breaking the documented behaviour that inverted decisions should take precedence over all others.
+    https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33671
+  - [`otelcol.exporter.kafka`,`otelcol.receiver.kafka`] Add config attribute to disable Kerberos PA-FX-FAST negotiation.
+    https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/26345
+  - [`OTTL`]: Added `keep_matching_keys` function to allow dropping all keys from a map that don't match the pattern.
+    https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32989
+  - [`OTTL`]: Add debug logs to help troubleshoot OTTL statements/conditions
+    https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33274
+  - [`OTTL`]: Introducing `append` function for appending items into an existing array.
+    https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32141
+  - [`OTTL`]: Introducing `Uri` converter parsing URI string into SemConv
+    https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32433
+  - [`OTTL`]: Added a Hex() converter function
+    https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33450
+  - [`OTTL`]: Added a IsRootSpan() converter function.
+    https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33729
+  - [`otelcol.processor.probabilistic_sampler`]: Add Proportional and Equalizing sampling modes.
+    https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31918
+  - [`otelcol.processor.deltatocumulative`]: Bugfix to properly drop samples when at limit.
+    https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33285
+  - [`otelcol.receiver.vcenter`] Fixes errors in some of the client calls for environments containing multiple datacenters.
+    https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33735
+  - [`otelcol.processor.resourcedetection`] Fetch CPU info only if related attributes are enabled.
+    https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33774
+  - [`otelcol.receiver.vcenter`] Adding metrics for CPU readiness, CPU capacity, and network drop rate.
+    https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33607
+  - [`otelcol.receiver.vcenter`] Drop support for vCenter 6.7.
+    https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33607
+  - [`otelcol.processor.attributes`] Add an option to extract value from a client address 
+    by specifying `client.address` value in the `from_context` field.
+    https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34048
+
+
```

Comment:
The routing keys supported in `otelcol.exporter.loadbalancing` are [pretty limited](https://github.com/grafana/alloy/blob/v1.2.1/internal/component/otelcol/exporter/loadbalancing/loadbalancing.go#L70-L88) right now.  This new routing key is for metrics. I decided not to add it since I'm not sure if the component will work correctly. It should be tested a bit more, and potentially [extra features](https://github.com/grafana/alloy/blob/v1.2.1/internal/component/otelcol/exporter/loadbalancing/loadbalancing.go#L35-L38) might need to be implemented.